### DISCUSSION TITLE
feat: UIs?

### DISCUSSION THREAD

**Discussion by NickCrews** [ibis-project/ibis#8404] [Ideas]
### Is your feature request related to a problem?

OK, this is really going out on a limb here, but I'm curious what your thoughts are.

I currently am using [solara](https://solara.dev/) to make little widgets, that I can use both in my jupyter notebooks, and host on websites. Would you be interested in some common widgets being integrated with ibis?

For example, I currently have something similar to https://motherduck.com/blog/introducing-column-explorer/:
<img width="978" alt="image" src="https://github.com/ibis-project/ibis/assets/10820686/85ac13a7-fc9a-4863-9143-4a904ebe031a">, that could be used something like `ibis.ui.profile(table)`

Or, a better table UI with filters, sorting, column stats built in.

I tried to make a no-code query builder a la https://react-querybuilder.js.org/, but that was proving hard to make it as flexible as I wanted, because I wanted to allow building arbitrary expressions. But a simple version of that would be possible.

### Describe the solution you'd like

Probably not part of this main repo, but could be a sister repo. And the ibis docs could reference it and use it in examples. Or just decide this isn't in scope, and maybe I'll get around to packaging that existing code into a community package that is totally external to ibis.

### What version of ibis are you running?

NA

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lostmygithubaccount** [comment]
we were recently discussing something like this: https://github.com/ydataai/ydata-profiling

I'm -1 on this in the main repo, but an experimental Ibis repo to build something like that seems cool to me?

**Comment by cpcloud** [comment]
@NickCrews Definitely interested in tight integration where possible. To be concrete, we're interested in other libraries being able to accept Ibis expressions directly and happy to help enable that.

I'll transfer this issue to a GitHub discussion, since I don't think it really makes sense to close it, and I'd like to keep this open for others to comment on.

**Comment by lostmygithubaccount** [comment]
@NickCrews any thoughts on whether it would make more sense to contribute into https://github.com/ydataai/ydata-profiling, allowing Ibis, versus a package specifically for Ibis?

**Reply by NickCrews** [reply]
yeah that looks like the right idea, [made an issue](https://github.com/ydataai/ydata-profiling/issues/1552)

**Comment by binste** [comment]
Cool idea with the profiling! I'd also find a "data catalog" explorer useful. Something like `show_catalog(ibis_connection)` or an interactive version of `ibis_connection.list_tables`. You'd then get an interactive view of all schemas, tables, views, ... in a database. When writing Ibis expressions, I often need something like DBeaver or the Snowflake Web UI open to see what kind of tables I have available. Is that something which could be interesting for the main repo?
